src/poetry/utils/env.py

```diff
@@ -1476,7 +1476,13 @@ def run_pip(self, *args: str, **kwargs: Any) -> int | str:
 
     def run_python_script(self, content: str, **kwargs: Any) -> int | str:
         return self.run(
-            self._executable, "-I", "-W", "ignore", "-", input_=content, **kwargs
+            self._executable,
+            "-W",
+            "ignore",
+            "-",
+            input_=content,
+            stderr=subprocess.DEVNULL,
```

Comment:
In the event that the command exits with a failure, will the output still be printed for the debug purposes?

Reply:
Right now the output isn't printed at all.

What I *can* do is modify the PR to instead get stderr separately, and log whatever's in it if it's non-empty. But this will make the PR more complex, and I would potentially be changing some function signatures, I'm not sure about the impact.

Reply:
I might have to do some local testing, but as far as I recall, if `run_python_script` fails we throw an EnvCommandError and the full output of the command is printed. An example is the issue report on https://github.com/python-poetry/poetry/issues/6627 -- we need to preserve this output for debugging/support reasons.

Reply:
Let me know how you want me to address this.

Reply:
I would suggest you store it, catch an exception from `subprocess`, and then throw a new exception containing stdout and stderr (with headers labeling each).

Reply:
WDYM catch an exception from `subprocess`? I don't think anything is thrown?

Reply:
There is an `except CalledProcessError` here that already catches and throws like I outlined -- that is where the output needs to be captured.

Reply:
Gotcha. Will look into this tomorrow. Thank you

Reply:
In fact, this data is already captured: https://docs.python.org/3/library/subprocess.html#subprocess.CalledProcessError

We just need to not merge it with stdout (or literally pip it to /dev/null) -- then it will be available on `CalledProcessError.stderr`.

Reply:
So you think reverting back to the default of `subprocess.PIPE` instead of `subprocess.DEVNULL` is enough? I can make the update now.

Reply:
I'm just providing review right now -- I don't have the time to do a close reading of the docs or write it myself at the moment.

The criteria for this change to be accepted are:
* does not mix stderr and stdout together
* preserves stderr and stdout on CalledProcessError
* EnvCommandError should print both stderr and stdout, and label which is which

Reply:
Im not sure that last bit is within the scope of this PR, but I've pushed an implementation of your suggestion as a follow-up commit, let me know what you think.

Reply:
It definitely is as the reason stdout and stderr were merged was so that `.output` could be used in that exception -- an unnecessary/silly kludge, but the intent is at least clear.